decidim-core/db/migrate/20240722215500_change_object_changes_on_versions.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class ChangeObjectChangesOnVersions < ActiveRecord::Migration[6.1]
+  def up
+    rename_column :versions, :object_changes, :old_object_changes
+    add_column :versions, :object_changes, :jsonb # or :json
+
+    PaperTrail::Version.where.not(old_object_changes: nil).find_each do |version|
+      # This is an adaptation of PaperTrail internal load_changeset method,having in mind that we
+      # need to call also the code from PaperTrail::AttributeSerializer::ObjectChangesAttribute
+      object_changes = ActiveSupport::HashWithIndifferentAccess.new(YAML.unsafe_load(version.old_object_changes))
+      unless version.item.nil?
+        # This is the deserialization code from `PaperTrail::AttributeSerializer::ObjectChangesAttribute`
+        # where we skip checking the object changeset column type, as we migrate it from YAML to JSON
+        changes_to_serialize = object_changes.clone
+        if changes_to_serialize.present?
+          serializer = PaperTrail::AttributeSerializers::CastAttributeSerializer.new(version.item.class)
+          changes_to_serialize.each do |key, change|
+            # `change` is an Array with two elements, representing before and after.
+            object_changes[key] = Array(change).map do |value|
+              serializer.send(:deserialize, key, value)
+            end
+          end
+        end
+      end
+
+      version.update_columns(old_object_changes: nil, object_changes:) # rubocop:disable Rails/SkipsModelValidations
+    end
+
```

Comment:
I think we can undo the change before the `db:seed` if we reset the column information cache. Try it out and let me know if it works:

```suggestion

    PaperTrail::Version.reset_column_information    
```